src/renderers/native/ReactNativeBridgeEventPlugin.js

```diff
@@ -65,6 +54,40 @@ var ReactNativeBridgeEventPlugin = {
     }
     return event;
   },
+
+  processEventTypes: function(
+    viewConfig: ReactNativeBaseComponentViewConfig,
+  ): void {
+    const {bubblingEventTypes, directEventTypes} = viewConfig;
+
+    if (__DEV__) {
+      for (const topLevelType in directEventTypes) {
+        invariant(
+          bubblingEventTypes[topLevelType] == null,
+          'Event cannot be both direct and bubbling: %s',
+          topLevelType,
+        );
+      }
+    }
+
+    for (const topLevelType in bubblingEventTypes) {
+      if (customBubblingEventTypes[topLevelType] == null) {
+        ReactNativeBridgeEventPlugin.eventTypes[
```

Comment:
I would expect that you'd need to call publishEventForPlugin in order for the event system to work properly. I guess we might only use the fields that populates (ex: EventPluginRegistry.possibleRegistrationNames) for DOM – but can we call publishEventForPlugin from this function too to make sure these are (mostly) going through the same paths as before and the same paths as if the event types are available on startup? Mostly in case we change how event registration works later and forget to update this.

It will validate that you don't supply the same event name twice but since you have the `customBubblingEventTypes[topLevelType] == null` check it should be OK I think.